### PR TITLE
Remove osmodel.mak in favor of ../dmd/osmodel.mak

### PR DESCRIPTION
**Pull request by wilzbach**
A small step at cleaning up the Makefiles.

Similarly to https://github.com/dlang/tools/pull/227

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ dependencies:
 
 test:
   override:
-    - make -f posix.mak style
     - ./circleci.sh setup-repos
+    - make -f posix.mak style
```
Comment (andralex) [inline circle.yml]: Is this related?

Comment (wilzbach) [inline circle.yml]: Yes, DMD is now needed before other Makefile tasks can be run.
